src/python/pants/backend/python/dependency_inference/rules.py

```diff
@@ -127,10 +127,7 @@ async def infer_python_dependencies(
         )
 
     owners_per_import = await MultiGet(owners_requests)
-    # We remove the request's address so that we don't infer dependencies on self.
-    merged_result = sorted(
-        set(itertools.chain.from_iterable(owners_per_import)) - {request.sources_field.address}
-    )
```

Comment:
What changed here? Is this just because it is safe for a target to depend on itself via a file address?

Reply:
It's leftover from when we inferred on the BUILD target, rather than file addresses. I figure it's never going to happen where you import your own file - that's not really legal Python to import yourself.  Dep inference is now _always_ at file level precision, so we also don't need to worry about inferring a dep on the owning BUILD target.